src/components/multi-select/multi-select.jsx

```diff
@@ -0,0 +1,316 @@
+  }, [value.map(val => JSON.stringify(val)).join(',')]);
+
+  useImperativeHandle(selfRef, () => ({
+    get dirty() {
+      return props.value.map(val => JSON.stringify(val)).join(',') !== this.value.map(val => JSON.stringify(val)).join(',');
+    },
+    id: props.id,
+    name: props.name,
+    get value() {
+      return value;
+    },
+  }));
+
+  return (
+    <div className={classNames.container} id={props.id} ref={wrapperRef}>
+      <FormControl
+        label={props.label}
+        labelId={ids.label}
+        id={ids.textbox}
+        error={validationMessage}
+        onKeyDown={onKeyDown}
+      >
+        <div role="presentation" className={classNames.formControlChildrenContainer} onClick={onClick}>
+          <TagList
+            className={classNames.tagList}
+            labelledBy={ids.label}
+            refs={valueRefs}
+          >
+            {value.length !== 0 &&
+              props.tagChildren ?
+              props.tagChildren(value, valueRefs, onOptionRemove) :
+              value.map((val, index) => (
+                <Tag
+                  defaultActive={index === 0}
+                  id={`${props.id}-option-${props.optionIDGetter(getOption(val))}`}
+                  key={props.optionIDGetter(getOption(val))}
+                  onRemove={onOptionRemove}
+                  readOnly={props.readOnly}
+                  ref={valueRefs[index]}
+                >
+                  {props.optionLabelGetter(
+                    props.options.find(option => props.optionIDGetter(option) === props.optionIDGetter(getOption(val))),
+                  )}
+                </Tag>
+              ))
+            }
+            <input
+              aria-autocomplete="list"
+              aria-controls={ids.listbox}
+              aria-describedby={`${ids.emptyMessage}`}
+              aria-expanded={expanded}
+              aria-haspopup="listbox"
+              aria-labelledby={ids.label}
+              autoComplete="off"
+              role="combobox"
+              className={classNames.input}
+              id={ids.textbox}
+              onBlur={onAutocompleteBlur}
+              onChange={onAutocompleteChange}
+              placeholder={value.length === 0 ? props.placeholder : undefined}
+              readOnly={props.readOnly}
+              required={props.required ? value.length === 0 : false}
+              ref={autocompleteRef}
+              value={autocompleteValue}
+            />
+          </TagList>
+          <FormControlIcons validationMessage={props.validationMessage} className={classNames.iconsContainer}>
+            {value.length > 0 && (
+              <IconButton
+                icon={iconClear}
+                label={`Remove all selected options on ${props.label}`}
+                onPress={onOptionsClear}
+                className={classNames.iconClear}
+              />
+            )}
+            <Icon icon={iconCaretDown} label="Open multi select options" />
```

Comment:
I know this wasn't implemented in the original multi-choice field, but I think we want an `IconButton` here for an interactive `MultiSelect` and an `Icon` for a read-only mode. 

In read-only mode, the label is incorrect as well. I wonder what's a good label for it? Maybe none since it's purely presentational?

Reply:
Made this change, but `Icon` requires a `label` attribute. LMK if you don't like the `Opening ${props.label} options disabled while read only` label I went with.